primitives/src/v2/mod.rs

```diff
@@ -136,97 +1628,176 @@ impl PvfCheckStatement {
 sp_api::decl_runtime_apis! {
 	/// The API for querying the state of parachains on-chain.
 	#[api_version(2)]
-	pub trait ParachainHost<H: Encode + Decode = v1::Hash, N: Encode + Decode = v1::BlockNumber> {
+	pub trait ParachainHost<H: Encode + Decode = Hash, N: Encode + Decode = BlockNumber> {
 		/// Get the current validators.
-		fn validators() -> Vec<v1::ValidatorId>;
+		fn validators() -> Vec<ValidatorId>;
 
 		/// Returns the validator groups and rotation info localized based on the hypothetical child
 		///  of a block whose state  this is invoked on. Note that `now` in the `GroupRotationInfo`
 		/// should be the successor of the number of the block.
-		fn validator_groups() -> (Vec<Vec<v1::ValidatorIndex>>, v1::GroupRotationInfo<N>);
+		fn validator_groups() -> (Vec<Vec<ValidatorIndex>>, GroupRotationInfo<N>);
 
 		/// Yields information on all availability cores as relevant to the child block.
 		/// Cores are either free or occupied. Free cores can have paras assigned to them.
-		fn availability_cores() -> Vec<v1::CoreState<H, N>>;
+		fn availability_cores() -> Vec<CoreState<H, N>>;
 
 		/// Yields the persisted validation data for the given `ParaId` along with an assumption that
 		/// should be used if the para currently occupies a core.
 		///
 		/// Returns `None` if either the para is not registered or the assumption is `Freed`
 		/// and the para already occupies a core.
-		fn persisted_validation_data(para_id: v1::Id, assumption: v1::OccupiedCoreAssumption)
-			-> Option<v1::PersistedValidationData<H, N>>;
+		fn persisted_validation_data(para_id: Id, assumption: OccupiedCoreAssumption)
+			-> Option<PersistedValidationData<H, N>>;
 
 		/// Returns the persisted validation data for the given `ParaId` along with the corresponding
 		/// validation code hash. Instead of accepting assumption about the para, matches the validation
 		/// data hash against an expected one and yields `None` if they're not equal.
 		fn assumed_validation_data(
-			para_id: v1::Id,
-			expected_persisted_validation_data_hash: v1::Hash,
-		) -> Option<(v1::PersistedValidationData<H, N>, v1::ValidationCodeHash)>;
+			para_id: Id,
+			expected_persisted_validation_data_hash: Hash,
+		) -> Option<(PersistedValidationData<H, N>, ValidationCodeHash)>;
 
 		/// Checks if the given validation outputs pass the acceptance criteria.
-		fn check_validation_outputs(para_id: v1::Id, outputs: v1::CandidateCommitments) -> bool;
+		fn check_validation_outputs(para_id: Id, outputs: CandidateCommitments) -> bool;
 
 		/// Returns the session index expected at a child of the block.
 		///
 		/// This can be used to instantiate a `SigningContext`.
-		fn session_index_for_child() -> v1::SessionIndex;
+		fn session_index_for_child() -> SessionIndex;
 
 		/// Old method to fetch v1 session info.
 		#[changed_in(2)]
-		fn session_info(index: v1::SessionIndex) -> Option<v1::SessionInfo>;
+		fn session_info(index: SessionIndex) -> Option<SessionInfo>;
```

Comment:
this should return v1::SessionInfo

Reply:
Really? That's bizarre. How are we ever supposed to remove v1 primitives?

Reply:
Oh, we should probably just remove this method as it's not going to be used anymore...